nemo/collections/nlp/models/language_modeling/megatron_t5_speechlm_model.py

```diff
@@ -1108,12 +1116,15 @@ def build_virtual_prompt_dataset(
             lm_vocab_size=self.lm_vocab_size,
             num_speech_codebooks=self.num_speech_codebooks,
             codebook_fps=self.cfg.data.get('codebook_fps', 75),
+            add_special_tokens_to_only_first_codebook=self.cfg.data.get('add_special_tokens_to_only_first_codebook', False),
         )
 
         rank = parallel_state.get_data_parallel_rank()
         world_size = parallel_state.get_data_parallel_world_size()
+        # generate a random seed to shuffle data differently in each run
+        seed = int(time.time())
         sampler = torch.utils.data.distributed.DistributedSampler(
-            dataset, num_replicas=world_size, rank=rank, shuffle=shuffle, seed=self.cfg.seed
+            dataset, num_replicas=world_size, rank=rank, shuffle=shuffle, seed=seed
```

Comment:
This is not a good idea. It should be fixed across different ranks. See https://pytorch.org/docs/stable/data.html#torch.utils.data.distributed.DistributedSampler

Reply:
I see. Sure, let me remove this.